encoding/descaling.md

```diff
@@ -1,477 +1,332 @@
+and the scaler used next to that.
+Make sure to set it to Nearest!
 
-```bash
-$ python getnative.py "descale_manaria04.png"
-```
+Figuring out what to look out for
+is a case of trial and error.
+The best thing to keep an eye on
+is lineart as well as small detail like stars.
+It isn't uncommon for noise to become weaker after a rescale
+(since that's usually added after upscaling, too),
+so try not to be too distracted by that.
 
-It should show the following:
+[descale]: https://github.com/Irrational-Encoding-Wizardry/vapoursynth-descale
+[fmtc]: https://forum.doom9.org/showthread.php?t=166504
 
-```
-Using imwri as source filter
-501/501
-Kernel: bicubic AR: 1.78 B: 0.33 C: 0.33
-Native resolution(s) (best guess): 878p
-done in 18.39s
-```
 
-If you check the directory
-where you executed the script,
-you will find a new folder
-called "getnative".
-You can find the following graph
-in there as well:
-
-![Manaria Friends — 01 (frame 4 getnative graph)](images/descale_graph.png)
-
-
-The X-axis shows the resolutions that were checked,
-and the Y-axis shows the relative error.
-The relative error refers to
-the difference between the original frame
-and the rescaled frame.
-What you're looking for are
-the spikes that show a low relative error.
-In this case it very clearly points to 878p.
-
-As a sidenote,
-it's important to keep in mind that this script
-can't find native 1080p elements.
-This is because it descales the frame
-and re-upscales it afterwards
-to determine the relative error.
-You can't descale to 1080p
-if the frame is already 1080p.
-If you have reason to believe
-that your show might be native 1080p,
-you've got to go with your gut.
-
-![Date A Live III — 01 (getnative graph)](images/descale_native1080_graph.png)
-
-An example of a graph for a native 1080p show.
-
-You may notice that the line swerves a bit
-in the first graph.
-There are going to be cases where
-you will get odd graphs like these,
-so it's important to know
-when the results are safe enough to descale
-or when they're too risky.
-Here is an example of
-a "bad" graph:
-
-![Miru Tights — 02 (getnative graph)](images/descale_bad_graph1.png)
+# Re-scaling
 
-```
-Output:
-Kernel: bicubic AR: 1.78 B: 0.33 C: 0.33
-Native resolution(s) (best guess): 869p, 848p
-```
+A fairly common practice is
+to upscale the descaled clip to a standard resolution whenever possible
+(if it isn't already one, like for example 720p).
+There's a couple scalers that are generally recommended.
 
-The script has determined that
-it's likely either 848p
-or 869p.
-However,
-there are no clear spikes in this graph
-like there was in the Manaria Friends one.
-The results are not clear enough
-to work off of.
-Here's another example:
+The most common scaler for re-scaling is [nnedi3_rpow2][].
+It returns consistently good-looking lines
+without damaging other detail.
+Since it's an image doubler,
+it's important to remember that you'll have to scale it down afterwards.
```

Comment:
No recommendation for a downscaling kernel?

Reply:
In the example code I use the internal scaler (which is Spline36), and it's also what I'd recommend either way. I'll note that Nnedi3_rpow2 (and resample) can downscale a clip inherently.